examples/dbapi_proxy_streaming.py

```diff
@@ -0,0 +1,43 @@
+import logging
+import asyncio
+import sqlite3
+from mysql_mimic import (
+    MysqlServer,
+    Session,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class DbapiProxyStreamingSession(Session):
+    def __init__(self):
+        super().__init__()
+        self.conn = sqlite3.connect(":memory:")
+
+    async def query(self, expression, sql, attrs):
+        cursor = self.conn.cursor()
+        cursor.execute(expression.sql(dialect="sqlite"))
+        if cursor.description:
+            column_names = [column[0] for column in cursor.description]
+            return self.fetch_data(cursor), column_names
+        return None
+
+    async def fetch_data(self, cursor):
+        while True:
+            rows = cursor.fetchmany(1000)
```

Comment:
In your actual use case, are you proxying to another engine using a synchronous DBAPI client?

If so, you might consider switching to an asyncio client, if one exists (depends on the engine you're proxying to). Without it, some of these calls (`execute`, `fetchmany`) will block the event loop while waiting on IO.

Reply:
We originally used pyodbc, since we did everything synchronoulsy before.

But aioodbc seems to be a good candidate for asynchronous odbc access. And it is built ontop of pyodbc.